javascript/testing/README.md

```diff
@@ -487,6 +491,65 @@ jest.mock('moment', ()=> ({
 }))
 ```
 
+## End-to-end Testing
+> For End-to-end testing we use [Cypress](https://www.cypress.io/) Framework. However, our suggested concepts could be adopted with other End-to-End frameworks/test runners. 
+
+The below diagram illustrates main blocks that involved in End-to-end test. 
+
+![End-to-end Test diagram](./resources/e2eDiagram.png)
+
+- **Test** 
+    - Uses the [AAA pattern](#aaa-pattern---arrange-act-assert)
+    - One and only one unit that __has assertions__ for testing components' integration and data integrity 
+    - Uses one or more `Page Objects` to access and interact with tested application page
```

Comment:
Uses one or more Flows to implement "act" part of the test.

Reply:
Maybe we should start with PageObject, then Flow and only then with Test, which uses Page Object and Flow?